CMakeLists.txt

```diff
@@ -68,32 +40,80 @@ set(savitar_HDRS
     ${CMAKE_CURRENT_BINARY_DIR}/src/SavitarExport.h
 )
 
-set(SAVITAR_VERSION 0.1.2)
-set(SAVITAR_SOVERSION 0)
-
-set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_FULL_LIBDIR})
+set(SAVITAR_VERSION 5.0.0)
+set(SAVITAR_SOVERSION 5)
 
 if(BUILD_STATIC)
     add_library(Savitar STATIC ${savitar_SRCS})
 else()
     add_library(Savitar SHARED ${savitar_SRCS})
 endif()
 
-set(Savitar_LINK_LIBRARIES pugixml)
-if(CMAKE_USE_PTHREADS_INIT)
-    list(APPEND Savitar_LINK_LIBRARIES pthread)
-endif()
-target_link_libraries(Savitar PUBLIC ${Savitar_LINK_LIBRARIES})
+set_project_standards(Savitar)
+use_threads(Savitar)
+set_rpath(TARGETS
+            Savitar
+          PATHS
+            "$<$<PLATFORM_ID:Linux>:usr/bin>"
+            "$<$<PLATFORM_ID:Linux>:usr/bin/lib>"
+            "$<$<PLATFORM_ID:Darwin>:../lib>"
+            "$<$<PLATFORM_ID:Darwin>:../Resources/lib>"
+           RELATIVE)
 
-if(NOT WIN32 OR CMAKE_COMPILER_IS_GNUCXX)
-    set_target_properties(Savitar PROPERTIES COMPILE_FLAGS -fPIC)
-endif()
+target_link_libraries(Savitar PUBLIC pugixml)
 
 if(BUILD_PYTHON)
-    set(SIP_EXTRA_FILES_DEPEND python/Types.sip python/MeshData.sip python/SceneNode.sip python/Scene.sip)
-    #set(SIP_EXTRA_SOURCE_FILES python/Types.cpp)
-    set(SIP_EXTRA_OPTIONS -g -n PyQt5.sip) # -g means always release the GIL before calling C++ methods. -n PyQt5.sip is required to not get the PyCapsule error
-    add_sip_python_module(Savitar python/ThreeMFParser.sip Savitar)
+    list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+    if(NOT DEFINED Python_VERSION)
+        set(Python_VERSION
+                3.10
+                CACHE STRING "Python Version" FORCE)
+        message(STATUS "Setting Python version to ${Python_VERSION}. Set Python_VERSION if you want to compile against an other version.")
+    endif()
+    if(APPLE)
+        set(Python_FIND_FRAMEWORK NEVER)
+    endif()
+    find_package(cpython ${Python_VERSION} QUIET COMPONENTS Interpreter Development)
+    if(NOT TARGET cpython::cpython)
+        find_package(Python ${Python_VERSION} EXACT REQUIRED COMPONENTS Interpreter Development)
+    else()
+        add_library(Python::Python ALIAS cpython::cpython)
+        set(Python_SITEARCH "${CMAKE_INSTALL_PREFIX}/lib/python3.10/site-packages")
```

Comment:
```suggestion
        set(Python_SITEARCH "${CMAKE_INSTALL_PREFIX}/lib/python${Python_VERSION}/site-packages")
```

Reply:
This basically applies when we use the Conan CPython package which doesn't set the Python_VERSION atm. So I will leave this be